src/buildblock/GeometryBlocksOnCylindrical.cxx

```diff
@@ -55,17 +56,32 @@ GeometryBlocksOnCylindrical::get_rotation_matrix(float alpha) const
                           stir::make_1d_array(0.F, -1 * std::sin(alpha), std::cos(alpha)));
 }
 
+void
+GeometryBlocksOnCylindrical::check_scanner_configuration(const Scanner& scanner)
+{
+  { // Ensure the number of axial crystals per bucket is a multiple of the number of rings
+    int num_axial_crystals_per_bucket = scanner.get_num_axial_crystals_per_block() * scanner.get_num_axial_blocks_per_bucket();
+    if (scanner.get_num_rings() % (num_axial_crystals_per_bucket) != 0)
+      {
+        error("Error in GeometryBlocksOnCylindrical: number of rings (%d) is not a multiple of the num_axial_crystals_per_bucket "
+              "(%d) = num_axial_crystals_per_block (%d) * num_axial_blocks_per_bucket (%d)",
+              scanner.get_num_rings(),
+              num_axial_crystals_per_bucket,
+              scanner.get_num_axial_crystals_per_block(),
+              scanner.get_num_axial_blocks_per_bucket());
```

Comment:
use the `boost::format` trick to avoid the deprecated form
```suggestion
        error(boost::format("Error in GeometryBlocksOnCylindrical: number of rings (%d) is not a multiple of the num_axial_crystals_per_bucket "
              "(%d) = num_axial_crystals_per_block (%d) * num_axial_blocks_per_bucket (%d)") %
              scanner.get_num_rings() %
              num_axial_crystals_per_bucket %
              scanner.get_num_axial_crystals_per_block() %
              scanner.get_num_axial_blocks_per_bucket());
```
(but ATM, apply this by hand and run `clang-format`)